Kickstarter-iOS/AppDelegate.swift

```diff
@@ -299,6 +299,28 @@ internal final class AppDelegate: UIResponder, UIApplicationDelegate {
     completionHandler(true)
   }
 
+  // MARK: - Functions
+
+  private func configureOptimizely(with key: String) {
+    let optimizelyClient = OptimizelyClient(sdkKey: key)
+
+    optimizelyClient.start { result in
+      var shouldUpdateClient: Bool
+
+      switch result {
+      case .success:
+        shouldUpdateClient = self.viewModel.inputs.optimizelyConfigured(isSuccess: true)
```

Comment:
I know we spoke about it but I'd like to still use a type here so that we don't have to do the switch outside of the view model. I think we can do this by extending Swift's `Result<T>` type:

```swift
extension Result where Failure == Optimizely.OptimizelyError {
  init(_ optimizelyResult: Optimizely.OptimizelyResult<Value>) {
    switch optimizelyResult {
    case .success(let value): self = .success(value)
    case .failure(let error): self = .failure(error)
    }
  }
}
```
^ we can add a simple test to just prove that this initializer returns the corresponding `Result<Value>` type for a given `Optimizely.OptimizelyResult<Value>` type.

and then this function can just be:
```swift
private func configureOptimizely(with key: String) {
  let optimizelyClient = OptimizelyClient(sdkKey: key)

  optimizelyClient.start { [weak self] result in
    if self?.viewModel.inputs.optimizelyConfigured(Result(result)) {
      print("🔮 Optimizely SDK Successfully Configured")
      AppEnvironment.updateOptimizelyClient(optimizelyClient)
    }
  }
}
```
^ I added `[weak self]` here too. Probably not that necessary considering we won't `deinit` our `AppDelegate` but 🤷‍♂.